### PR TITLE
moved voxel plot to api accessible function

### LINKED ISSUE #2219: Feature request: voxel plots from within the python API
Everyone loves a nice voxel plot 

:black_square_button: :red_square: :red_square: :red_square: :black_square_button: 
 :red_square: :red_square: :red_square:  :red_square: :red_square:
 :black_square_button: :black_square_button: :black_square_button:  :red_square: :red_square:
 :brown_square: :brown_square: :black_square_button:  :red_square: :red_square:
:black_square_button: :brown_square: :black_square_button: :red_square: :black_square_button: 

The current method requires a mixture of the python API and running the ```openmc-voxel-to-vtk``` script.

```python
vox_plot = openmc.Plot()
vox_plot.type = 'voxel'
vox_plot.width = (1500., 1500., 1500.)
vox_plot.pixels = (200, 200, 200)
vox_plot.filename = 'output_file'   # used for the h5 filename
plots = openmc.Plots([vox_plot])
plots.export_to_xml()

openmc.plot_geometry()
os.system('openmc-voxel-to-vtk output_h5_file.h5 -o plot_3d_tokamak.vti')
```

I would be keen to move some of the functionally into the the Python API so that the voxel vti file can be make entirely from within the Python API. This could also be accessed from the script so we wouldn't lose functionality. Similar to the #2171 PR where we moved the track writing from the script file to the Python track object. The above user code would then look a bit more like this ...

```python
vox_plot = openmc.Plot()
vox_plot.type = 'voxel'
vox_plot.width = (1500., 1500., 1500.)
vox_plot.pixels = (200, 200, 200)
vox_plot.filename = 'output_file'   # used for the h5 and vti filename
plots = openmc.Plots([vox_plot])
plots.export_to_xml()

openmc.plot_geometry()  # produces the vti file as well as the h5 file
```

Having suggested this movement of code I'm not entirely sure how to go about it but I'm taking a look at the openmc.plot_geometry and I'm sure it would be possible. Any comments on if this would be a welcome improvement or hints about how this could be done are most welcome.

### PR DESCRIPTION
**Pull request by shimwell**
This issue moves some voxel plotting code from a python script to the python api

The script retains it's original functionality as it calls the python api

The benefit is that this allows one to make a voxel plot from the python api as well as running the conversion script form the terminal.

before this PR
```python
import os
import openmc
surf1 = openmc.Sphere(r=500, boundary_type='vacuum')
cell1 = openmc.Cell(region=-surf1)
geometry = openmc.Geometry([cell1])
geometry.export_to_xml()
materials = openmc.Materials()
materials.export_to_xml()
vox_plot = openmc.Plot()
vox_plot.type = 'voxel'
vox_plot.id = 12
vox_plot.width = (1500., 1500., 1500.)
vox_plot.pixels = (200, 200, 200)
vox_plot.color_by = 'cell'
openmc.plot_geometry()
os.system('openmc-voxel-to-vtk plot_12.h5 -o test_voxel_plot.vti')
```
after this PR
```python
import openmc
surf1 = openmc.Sphere(r=500, boundary_type='vacuum')
cell1 = openmc.Cell(region=-surf1)
geometry = openmc.Geometry([cell1])
geometry.export_to_xml()
materials = openmc.Materials()
materials.export_to_xml()
vox_plot = openmc.Plot()
vox_plot.type = 'voxel'
vox_plot.id = 12
vox_plot.width = (1500., 1500., 1500.)
vox_plot.pixels = (200, 200, 200)
vox_plot.color_by = 'cell'
vox_plot.to_voxel_file(output='test_voxel_plot.vti')
```
closes #2219

### CODE REVIEW

**Review by paulromano — CHANGES_REQUESTED** [review]
Thanks for moving this code over to the proper API @shimwell. I spruced it up a bit and sent a PR to your fork: https://github.com/shimwell/openmc/pull/29

File: openmc/plots.py [hunk 1]
```diff
@@ -178,6 +181,76 @@ def _get_plot_image(plot, cwd):
     return Image(str(png_file))
 
 
+def voxel_to_vtk(voxel_file: PathLike, output: PathLike = 'plot.vti'):
+    """Converts a voxel HDF5 file to a VTK file
+
+    Parameters
```
Comment (paulromano) [inline openmc/plots.py:187]:
```suggestion
    """Converts a voxel HDF5 file to a VTK file

    .. versionadded:: 0.13.4

    Parameters
```

File: openmc/plots.py [hunk 2]
```diff
@@ -831,6 +904,43 @@ def to_ipython_image(self, openmc_exec='openmc', cwd='.'):
         # Return produced image
         return _get_plot_image(self, cwd)
 
+    def to_vtk(self, output: Optional[PathLike] = None,
+               openmc_exec: str = 'openmc', cwd: str = '.'):
+        """Render plot as an voxel image
+
+        This method runs OpenMC in plotting mode to produce a .vti file.
+
+        Parameters
```
Comment (paulromano) [inline openmc/plots.py:913]:
```suggestion
        This method runs OpenMC in plotting mode to produce a .vti file.

        .. versionadded:: 0.13.4

        Parameters
```

File: scripts/openmc-voxel-to-vtk [hunk 3]
```diff
@@ -73,4 +17,6 @@ if __name__ == "__main__":
         help="Path to output VTK file.",
     )
     args = parser.parse_args()
-    voxel_to_vtk(args.voxel_file, args.output)
+    print("Reading and translating data...")
+    openmc.plots._voxel_to_vtk(args.voxel_file, args.output)
```
Comment (paulromano) [inline scripts/openmc-voxel-to-vtk:21]:
```suggestion
    openmc.voxel_to_vtk(args.voxel_file, args.output)
```

**Review by paulromano — CHANGES_REQUESTED** [review]
Oops, forgot some `versionadded` directives and noticed we needed an update in the script

**Review by paulromano — APPROVED** [review]

**Comment by shimwell** [comment]
If anyone is interested in an example of this feature I just made one over [here](https://github.com/firstlightfusion/neutronics-workshop/blob/main/tasks/task_03_making_CSG_geometry/3_viewing_the_geometry_in_3d.ipynb)

**Pull request merged**